flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java

```diff
@@ -649,16 +649,32 @@ public void sqlUpdate(String stmt) {
 	private TableResult executeOperation(Operation operation) {
 		if (operation instanceof CreateTableOperation) {
 			CreateTableOperation createTableOperation = (CreateTableOperation) operation;
-			catalogManager.createTable(
-					createTableOperation.getCatalogTable(),
-					createTableOperation.getTableIdentifier(),
-					createTableOperation.isIgnoreIfExists());
+			if (createTableOperation.isTemporary()) {
+				catalogManager.createTemporaryTable(
+						createTableOperation.getCatalogTable(),
+						createTableOperation.getTableIdentifier(),
+						createTableOperation.isIgnoreIfExists());
+			} else {
+				catalogManager.createTable(
+						createTableOperation.getCatalogTable(),
+						createTableOperation.getTableIdentifier(),
+						createTableOperation.isIgnoreIfExists());
+			}
 			return TableResultImpl.TABLE_RESULT_OK;
 		} else if (operation instanceof DropTableOperation) {
 			DropTableOperation dropTableOperation = (DropTableOperation) operation;
-			catalogManager.dropTable(
-					dropTableOperation.getTableIdentifier(),
-					dropTableOperation.isIfExists());
+			if (dropTableOperation.isTemporary()) {
+				boolean dropped = catalogManager.dropTemporaryTable(dropTableOperation.getTableIdentifier());
```

Comment:
I would suggest to extend `catalogManager.dropTemporaryTable` to take care of the logic about `ifExists`, be consistent with `catalogManager.dropTable`

Reply:
`TableEnvironment#dropTemporaryTable` and `dropTemporaryView` also return `boolean` which is consistent with `CatalogManager#dropTemporaryTable` and `dropTemporaryView`. Shall we refactor these interfaces in another ticket?

Reply:
Since we are talking about CatalogManager's interface, I think it's make sense to compare the interfaces belong to CatalogManager. It looks a bit strange that some of the 'ignoreIfNotExists' logic is happened inside CatalogManager, but some of them are handled out of CatalogManager. I would suggest to keep it consistent.

Reply:
OK. I can refactor the CatalogManager first.